xls/dslx/stdlib/std.x

```diff
@@ -177,6 +177,53 @@ fn iterative_div_test () {
   ()
 }
 
+// Binary Division algorithm as outlined by:
+// http://www.dragonwins.com/domains/GetTechEd/de248/binary_division.htm
```

Comment:
should we link to https://github.com/google/xls/blob/fa7af6cbda69d02d2d50b28fa5d95f51fe8cdfb1/xls/ir/abstract_evaluator.h#L513-L581 now?

Reply:
Probably not a bad idea to make the link explicit, yeah.

Also probably worth adding a comment (like I did in the AbstractEvaluator) calling out that this is a slow operation to do combinationally, and might be worth pipelining in some circumstances.

As a separate followup, it's probably worth it for us to verify which synthesizes better in practice for combinational implementations at different sizes: this, restoring division (a really small modification, just replacing (r >= d) with an unconditional subtract, test if negative, & adding d back if the result is negative), or non-restoring division (per https://en.wikipedia.org/wiki/Division_algorithm#Non-restoring_division).

Reply:
do we want to support the 3 variants with parametrics?

Reply:
Interesting thought. I think for reasonable ease of use, that would mean ~parametrizing on an *enum*... and I'm not sure we can do that. Let's go ahead with this as-is for now, and consider our options later.

Reply:
I think it would work (although we might have to have additional wrapper functions to address partial specialization and default value); I'd be curious if the IR optimization pass would be capable of optimization the constant and the match expression away if we made it a regular parameter.

Reply:
> consider our options later

filed #1046

Reply:
updated docstring to mention the original IR implementation and current limitations.